File: src/test/java/handler/InjuryHandlerTest.java
package handler;

import collect.misc.Injuries;
import handler.InjuryHandler;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.HashMap;
import java.util.Map;

import static org.mockito.Mockito.verify;

class InjuryHandlerTest {

    @Mock
    Injuries injuries;

    @InjectMocks
    InjuryHandler injuryHandler;

    @BeforeEach
    void setUp() {
        MockitoAnnotations.initMocks(this);
    }

    @Test
    void handleMLBRequest() {
        Map<String, String> testMap = new HashMap<>();
        testMap.put("sport", "mlb");
        injuryHandler.handleRequest(testMap);
        verify(injuries).getStandardInjuryData("mlb");
    }

    @Test
    void handleNFLRequest() {
        Map<String, String> testMap = new HashMap<>();
        testMap.put("sport", "nfl");
        injuryHandler.handleRequest(testMap);
        verify(injuries).getNFLInjuryData();
    }

    @Test
    void handleNBARequest() {
        Map<String, String> testMap = new HashMap<>();
        testMap.put("sport", "nba");
        injuryHandler.handleRequest(testMap);
        verify(injuries).getStandardInjuryData("nba");
    }

    @Test
    void handleNHLRequest() {
        Map<String, String> testMap = new HashMap<>();
        testMap.put("sport", "nhl");
        injuryHandler.handleRequest(testMap);
        verify(injuries).getStandardInjuryData("nhl");
    }
}

File: src/test/kotlin/handler/InjuryHandlerTest.kt
package handler

import io.mockk.every
import io.mockk.spyk
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class InjuryHandlerTest {
    private val injuryHandler = spyk<InjuryHandler>()
    private val sport = "not nfl"
    private val nflResult: Map<String, String> = mapOf("nfl" to "result")
    private val standardResult: Map<String, String> = mapOf("standard" to "result")

    @BeforeEach
    fun setUp() {
        every { injuryHandler.getNFLInjuryData() } returns nflResult
        every { injuryHandler.getStandardInjuryData(sport) } returns standardResult
    }

    @Test
    fun `should handle request for nfl injury data`() {
        val result = injuryHandler.handleRequest(mapOf("sport" to "nfl"))
        assertEquals(nflResult, result)
    }

    @Test
    fun `should handle request with pipeline invocation`() {
        val result = injuryHandler.handleRequest(mapOf("sport" to sport))
        assertEquals(standardResult, result)
    }
}